search/search_quality/features_collector_tool/features_collector_tool.cpp

```diff
@@ -0,0 +1,185 @@
+#include "search/result.hpp"
+#include "search/search_quality/sample.hpp"
+#include "search/search_tests_support/test_search_engine.hpp"
+#include "search/search_tests_support/test_search_request.hpp"
+#include "search/v2/ranking_info.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "indexer/feature_algo.hpp"
+
+#include "platform/local_country_file.hpp"
+#include "platform/local_country_file_utils.hpp"
+#include "platform/platform.hpp"
+
+#include "base/string_utils.hpp"
+
+#include "std/fstream.hpp"
+#include "std/iostream.hpp"
+#include "std/limits.hpp"
+#include "std/string.hpp"
+#include "std/unique_ptr.hpp"
+#include "std/vector.hpp"
+
+#include "3party/gflags/src/gflags/gflags.h"
+
+using namespace search;
+using namespace search::tests_support;
+
+DEFINE_string(data_path, "", "Path to data directory (resources dir)");
+DEFINE_string(mwm_path, "", "Path to mwm files (writable dir)");
+DEFINE_string(json_in, "", "Path to the json file with samples (default: stdin)");
+
+struct Context
+{
+  Context(Index & index) : m_index(index) {}
+
+  void GetFeature(FeatureID const & id, FeatureType & ft)
+  {
+    auto const & mwmId = id.m_mwmId;
+    if (!m_guard || m_guard->GetId() != mwmId)
+      m_guard = make_unique<Index::FeaturesLoaderGuard>(m_index, mwmId);
+    m_guard->GetFeatureByIndex(id.m_index, ft);
+  }
+
+  Index & m_index;
+  unique_ptr<Index::FeaturesLoaderGuard> m_guard;
+};
+
+void GetContents(istream & is, string & contents)
+{
+  string line;
+  while (getline(is, line))
+  {
+    contents.append(line);
+    contents.push_back('\n');
+  }
+}
+
+bool Matches(Context & context, Sample::Result const & golden, search::Result const & actual)
+{
+  static double const kEps = 2 * 1e-5;
```

Comment:
I just do not understand why do we need to write 'constexpr instead of const for locals' everywhere.